### PR TITLE
Yes no boolean refactor

### PR DESCRIPTION
**Pull request by ratneshraval**
#### Which User story does this relate to?
- :(

#### Brief feature description
- RFA 01 C updated with YesNoComponent to use true boolean 
- RFA 01 A address card also updated to reflect how we can use boolean there.

### CODE REVIEW

File: app/javascript/rfa_forms/rfa01a_edit_view/addressCard.js [hunk 1]
```diff
@@ -62,6 +62,18 @@ export default class AddressCard extends React.Component {
     this.props.setParentState('addresses', data.toJS())
   }
 
+  handleClearOnConditionalChange (typeString, key, value, hiddenKey, hiddenDefaultValue) {
```
Comment (lauherkcambria) [inline app/javascript/rfa_forms/rfa01a_edit_view/addressCard.js]: do we need this method here?

Comment (ratneshraval) [reply app/javascript/rfa_forms/rfa01a_edit_view/addressCard.js]: No, that got in from a merge. I'll remove it.

File: app/javascript/rfa_forms/rfa01a_edit_view/addressCard.js [hunk 2]
```diff
@@ -94,10 +94,12 @@ export default class AddressCard extends React.Component {
...
                 idPrefix='mailing_similar'
-                value={mailingAddress}
-                onFieldChange={(event) => this.props.handleClearOnConditionalChange('physical_mailing_similar', 'addresses', event.target.value, [blankMailingAddress])} />
+                value={this.props.physicalMailingSimilar}
+                onFieldChange={(selectedValue) => this.props.handleClearOnConditionalChange('physical_mailing_similar', 'addresses', selectedValue, [blankMailingAddress])}
+              />
             </div>
-            <div className={hiddenMailingSameAsPhysical}>
+
+            {this.props.physicalMailingSimilar && !this.props.physicalMailingSimilar &&
```
Comment (lauherkcambria) [inline app/javascript/rfa_forms/rfa01a_edit_view/addressCard.js]: i'm a little confused by the logic on this line, can you explain?

Comment (ratneshraval) [reply app/javascript/rfa_forms/rfa01a_edit_view/addressCard.js]: No worries. The logic is simply
`this.props.physicalMailingSimilar && !this.props.physicalMailingSimilar` :)
Did that help?

Comment (ratneshraval) [reply app/javascript/rfa_forms/rfa01a_edit_view/addressCard.js]: The first part is to handle `undefined` and the 2nd part is to handle boolean part?

**Pull request closed without merging**